frameworks/kafka/docs/security.md

```diff
@@ -0,0 +1,245 @@
+
+#### Prerequisites
+- Completion of the section [Transport Encryption](#transport-encryption) above
+
+#### Install the Service
+
+Install the DC/OS Apache Kafka service with the following options in addition to your own:
+```json
+{
+    "service": {
+        "service_account": "<service-account>",
+        "service_account_secret": "<secret path>",
+        "security": {
+            "transport_encryption": {
+                "enabled": true
+            },
+            "ssl_authentication": {
+                "enabled": true
+            }
+        }
+    }
+}
+```
+
+*Note*: It is possible to enable SSL Auth after initial installation, but the service may be unavailable during the transition. Additionally, your Kafka clients will need to be reconfigured.
+
+#### Authenticating a Client
+
+To authenticate a client against DC/OS Apache Kafka, you will need to configure it to use a certificate signed by the DC/OS CA. After generating a certificate signing request, you can issue it to the DC/OS CA by calling the API `<dcos-cluster>/ca/api/v2/sign`. Using the `curl` command the request would look like:
+```bash
+$ curl -X POST \
+    -H "Authorization: token=$(dcos config show core.dcos_acs_token)" \
+    <dcos-cluster>/ca/api/v2/sign \
+    -d '{"certificate_request": "<json-encoded-value-of-request.csr>"}'
+```
+
+The response will contain a signed public certificate. Full details on the DC/OS CA API can be found [here](https://docs.mesosphere.com/latest/security/ent/tls-ssl/ca-api/).
+
+## Authorization
+
+The DC/OS Apache Kafka Service supports Kafka's ACL-based authorization (authz) system. To use Kafka's authz, either SSL Auth or Kerberos must be enabled as detailed above.
+
+### Enable Authorization
+
+#### Prerequisites
+- Completion of either [SSL Auth](#ssl-auth) or [Kerberos](#kerberos-authn) above.
+
+#### Install the Service
+
+Install the DC/OS Apache Kafka service with the following options in addition to your own (remember, either SSL Auth or Kerberos _must_ be enabled):
+```json
+{
+    "service": {
+        "security": {
+            "authorization": {
+                "enabled": true,
+                "super_users": "<list of super users>",
+                "allow_everyone_if_no_acl_found": <true|false default false>
+            }
+        }
+    }
+}
+```
+
+`service.security.authorization.super_users` should be set to a semi-colon delimited list of principals to treat as super users (all permissions). The format of the list is `User:<user1>;User:<user2>;...`. Using Kerberos authentication, the "user" value is the Kerberos primary, and for SSL Auth the "user" value is the `CN` of the certificate. The Kafka brokers themselves are automatically designated as super users.
```

Comment:
(We could think about adding support for `sasl.kerberos.principal.to.local.rules`: http://kafka.apache.org/documentation/#security_authz so that this is simpler)

https://jira.mesosphere.com/browse/INFINITY-3099